### PR TITLE
Use Java16 `Stream::toList`

### PR DESCRIPTION
**Pull request by tbenr**
I translated `.collect(Collectors.toUnmodifiableList())` and `.collect(Collectors.toList())` to `.toList()`

The assumption is that whenever we used `.collect(Collectors.toList())` was in any case intended to be an immutable list.

https://blogs.oracle.com/javamagazine/post/the-hidden-gems-in-java-16-and-java-17-from-streammapmulti-to-hexformat

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java [hunk 1]
```diff
@@ -399,15 +398,15 @@ private <T> SafeFuture<T> makeRequestToFailoversUntilSuccess(
...
+    final Iterator<? extends RemoteValidatorApiChannel> failoverDelegates =
         beaconNodeReadinessManager.getFailoversInOrderOfReadiness();
     return makeRequestToFailoversUntilSuccess(
         failoverDelegates.next(), failoverDelegates, request, method, capturedExceptions);
   }
 
   private <T> SafeFuture<T> makeRequestToFailoversUntilSuccess(
       final RemoteValidatorApiChannel currentFailoverDelegate,
-      final Iterator<RemoteValidatorApiChannel> failoverDelegates,
+      final Iterator<? extends RemoteValidatorApiChannel> failoverDelegates,
```
Comment (StefanBratanov) [inline validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java:409]: Is there a reason for having few changes of those `? extends`

**Review by StefanBratanov — APPROVED** [review]
LGTM apart from a question

File: validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java (hunk 1 above)
Comment (tbenr) [reply validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java:409]: Yes, the root of it is
```java
final List<? extends RemoteValidatorApiChannel> failoverValidatorApis =
    failoverEndpoints.stream()
        .map(
            endpoint ->
                RemoteValidatorApiHandler.create(
                    endpoint,
                    okHttpClient,
                    spec,
                    validatorConfig.isValidatorClientUseSszBlocksEnabled(),
                    asyncRunner))
        .toList();
```

Comment (tbenr) [reply validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java:409]: This is not the only place it happens, when we want to use the higher level interface as type

Comment (StefanBratanov) [reply validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java:409]: Got it. I don't mind it. Makes sense.

**Pull request merged**